concent_api/core/transfer_operations.py

```diff
@@ -102,33 +99,64 @@ def store_pending_message(
     )
 
 
-def create_file_transfer_token(
+def create_file_transfer_token_for_concent(
+    report_computed_task: message.tasks.ReportComputedTask,
+    authorized_client_public_key: bytes,
+    operation: FileTransferToken.Operation,
+) -> FileTransferToken:
+    ten_minutes = 600
+    return _create_file_transfer_token(report_computed_task, authorized_client_public_key, operation, ten_minutes)
+
+
+def create_file_transfer_token_for_golem_client(
     report_computed_task:       message.tasks.ReportComputedTask,
     authorized_client_public_key: bytes,
     operation:                  FileTransferToken.Operation,
+) -> FileTransferToken:
+    return _create_file_transfer_token(report_computed_task, authorized_client_public_key, operation)
+
+
+def _create_file_transfer_token(
+    report_computed_task: message.tasks.ReportComputedTask,
+    authorized_client_public_key: bytes,
+    operation: FileTransferToken.Operation,
+    deadline: Optional[int] = None
+
 ) -> FileTransferToken:
     """
     Function to create FileTransferToken from ReportComputedTask message
     """
+    def calculate_token_expiration_deadline(
+        operation: FileTransferToken.Operation,
+        deadline: Optional[int],
+    ) -> int:
+        if deadline is not None:
+            token_expiration_deadline = get_current_utc_timestamp() + deadline
+        else:
+            if operation == FileTransferToken.Operation.upload:
+                token_expiration_deadline = (
+                        report_computed_task.task_to_compute.compute_task_def['deadline'] +
+                        3 * settings.CONCENT_MESSAGING_TIME +
+                        2 * settings.MAXIMUM_DOWNLOAD_TIME
+                )
+
+            elif operation == FileTransferToken.Operation.download:
+                token_expiration_deadline = (
+                        report_computed_task.task_to_compute.compute_task_def['deadline'] +
+                        settings.SUBTASK_VERIFICATION_TIME
+                )
+        return token_expiration_deadline
+
     task_id         = report_computed_task.task_to_compute.compute_task_def['task_id']
     subtask_id      = report_computed_task.task_to_compute.compute_task_def['subtask_id']
     file_path       = get_storage_file_path(task_id, subtask_id)
 
     assert isinstance(authorized_client_public_key, bytes)
+    assert isinstance(deadline, int) or deadline is None
```

Comment:
Check this out:
```
>>> deadline = False
>>> isinstance(deadline, int) or deadline is None
True
```

Reply:
normal thing for python - it treats `bool` values as `int` values. It's not a bug, it's a feature ;)

Reply:
done

Reply:
I know, but this value will pass this assert.